src/controller.rs

```diff
@@ -1216,31 +1221,101 @@ impl GooseAttack {
+                                        self.reply_to_controller(
+                                            message,
+                                            GooseControllerResponseMessage::Bool(true),
+                                        );
+                                    }
+                                    // If the load test is running, rebuild the active test plan.
+                                    AttackPhase::Increase
+                                    | AttackPhase::Decrease
+                                    | AttackPhase::Maintain => {
+                                        info!(
+                                            "changing users from {} to {}",
+                                            goose_attack_run_state.active_users, new_users
+                                        );
+                                        // Determine how long has elapsed since this step started.
+                                        let elapsed = self.step_elapsed() as usize;
+
+                                        // Determine how quickly to adjust user account.
+                                        let hatch_rate = if let Some(hatch_rate) =
+                                            self.configuration.hatch_rate.as_ref()
+                                        {
+                                            util::get_hatch_rate(Some(hatch_rate.to_string()))
+                                        } else {
+                                            util::get_hatch_rate(None)
+                                        };
+                                        // Convert hatch_rate to milliseconds.
+                                        let ms_hatch_rate = 1.0 / hatch_rate * 1_000.0;
+                                        // Determine how many users to increase or decrease by.
+                                        let user_difference = (goose_attack_run_state.active_users
+                                            as isize
+                                            - new_users as isize)
+                                            .abs();
+                                        // Multiply the user difference by the hatch rate to get the total_time required.
+                                        let total_time =
+                                            (ms_hatch_rate * user_difference as f32) as usize;
+
+                                        // Reset the test_plan to adjust to the newly specified users.
+                                        self.test_plan.steps = vec![
+                                            // Record how many active users there are currently.
+                                            (goose_attack_run_state.active_users, elapsed),
+                                            // Configure the new user count.
+                                            (new_users, total_time),
+                                        ];
```

Comment:
Won't this remove history if there were multiple steps before this change?

Reply:
No, history is stored in `self.metrics.history` -- we do this same trick in [cancel_attack()](https://github.com/tag1consulting/goose/blob/main/src/lib.rs#L1728).

Reply:
(That's part of why I showed the Overview table in the book page: that's the actual history shown after executing this controller command multiple times.)